src/main/java/com/clevertap/stormdb/StormDB.java

```diff
@@ -502,6 +502,18 @@ public void put(int key, byte[] value, int valueOffset) throws IOException {
         rwLock.writeLock().lock();
 
         try {
+            boolean updatedInPlace = false;
+
+            final int recordIndexForKey = index.get(key);
+
+            if ((recordIndexForKey != RESERVED_KEY_MARKER) && ((isCompactionInProgress() && compactionState.dataInNextWalFile.get(key)) || (!isCompactionInProgress() && dataInWalFile.get(key)))) { // It means key exist
+                long address =  RecordUtil.indexToAddress(recordSize, recordIndexForKey);
+                if (address >= bytesInWalFile) { // i.e data is in the buffer
```

Comment:
```suggestion
                // Is the record in the buffer?
                if (address >= bytesInWalFile) {
```

Reply:
Will there be a case where `address == bytesInWalFile`? Can we simplify it to `address > bytesInWalFile`?

Reply:
I kept it consistent with randomGet function. There we are using >= itself

Reply:
Got it. Thanks!